internal/rbd/rbd_util.go

```diff
@@ -1165,13 +1225,42 @@ func (rv *rbdVolume) resize(newSize int64) error {
 	}
 	defer image.Close()
 
+	thick, err := rv.isThickProvisioned()
+	if err != nil {
+		return err
+	}
+
+	// offset is used to track from where on the expansion is done, so that
+	// the extents can be allocated in case the image is thick-provisioned
+	var offset uint64
+	if thick {
+		st, statErr := image.Stat()
+		if statErr != nil {
+			return statErr
+		}
+
+		offset = st.Size
+	}
+
 	err = image.Resize(uint64(util.RoundOffVolSize(newSize) * helpers.MiB))
 	if err != nil {
 		return err
 	}
 
+	if thick {
+		err = rv.allocate(offset)
+		if err != nil {
+			resizeErr := image.Resize(uint64(util.RoundOffVolSize(newSize) * helpers.MiB))
+			if resizeErr != nil {
+				err = fmt.Errorf("failed to shrink image (%v) after failed allocation: %w", resizeErr, err)
```

Comment:
failed to shrink the image? as we don't support shrinking do we need to update it?

Reply:
If writing to the RBD image failed, resizing is not complete. So in this case, the additional zeros that have been written (partially) can be chopped off again. There is no filesystem data on the new extension, so reducing the size is safe.